Docs for SVM in new interfaces (#1323)
oneapi-src/oneDAL: docs/source/onedal/kernel-functions/linear-kernel.rst

.. |c_result| replace:: `compute_result <linear_kernel_c_api_result_>`_
.. |c_op| replace::     `compute(...) <linear_kernel_c_api_>`_

=============== ============= ======== =========== ==================
 **Operation**  **Computational methods**   **Programming Interface**
--------------- --------------------------- -------------------------
   |c_math|      |c_dense|     |c_op|   |c_input|          |c_result|
=============== ============= ======== =========== ==================

------------------------
Mathematical formulation
------------------------

.. _linear_kernel_c_math:

Computing
---------

Given a set :math:`X` of :math:`n` feature vectors :math:`x_1 = (x_{11}, \ldots, x_{1p}), 
\ldots, x_n = (x_{n1}, \ldots, x_{np})` of dimension :math:`p` and a set :math:`Y` of :math:`m`
feature vectors :math:`y_1 = (y_{11}, \ldots, y_{1p}), \ldots, y_m = (y_{m1}, \ldots, x_{mp})`, 
the problem is to compute the linear kernel function :math:`K(x_i,, y_i)` for
any pair of input vectors: :math:`K(x_i, y_i) = k {X_i}^T y_i + b`.

.. _linear_kernel_c_math_dense:

---------------------
Programming Interface
---------------------

> All types and functions in this section shall be declared in the
> ``oneapi::dal::linear_kernel`` namespace and be available via inclusion of the
> ``oneapi/dal/algo/linear_kernel.hpp`` header file.

Reviewer: ```suggestion
All types and functions in this section are declared in the
``oneapi::dal::linear_kernel`` namespace and are available via inclusion of the
``oneapi/dal/algo/linear_kernel.hpp`` header file.
```
